File: experiments/rule-candidates/python/method-name-promises/fixtures/repository.py
"""A repository of cart rows, in memory."""


class CartRepository:
    def __init__(self) -> None:
        self.rows: dict[str, int] = {}
        self.audit_log: list[str] = []

    def save(self, row_id: str, cents: int) -> int:
        self.rows[row_id] = cents
        return len(self.rows)

    def find(self, row_id: str) -> int | None:
        return self.rows.get(row_id)

    def count(self) -> list[str]:
        return list(self.rows)

    def has(self, row_id: str) -> str:
        return "yes" if row_id in self.rows else "no"

    def clear(self) -> int:
        seen = 0
        for _ in self.rows:
            seen += 1
        return seen

    def total_cents(self) -> int:
        return sum(self.rows.values())

    def remove(self, row_id: str) -> bool:
        if row_id in self.rows:
            del self.rows[row_id]
            return True
        return False

    def audit(self, message: str) -> None:
        self.audit_log.append(message)

    def load_all(self, rows: dict[str, int]) -> None:
        self.rows.update(rows)

    def find_or_create(self, row_id: str) -> int:
        if row_id in self.rows:
            return self.rows[row_id]
        self.rows[row_id] = 0
        return 0

    def validate(self, row_id: str, cents: int) -> list[str]:
        errors = []
        if cents < 0:
            errors.append("cents must not be negative")
        self.rows[row_id] = cents
        return errors

File: rules/typescript/method-name-promises/fixtures/repository.ts
export interface Row {
  id: string;
  cents: number;
}

export class CartRepository {
  private rows = new Map<string, Row>();
  private auditLog: string[] = [];

  save(row: Row): number {
    this.rows.set(row.id, row);
    return this.rows.size;
  }

  find(id: string): Row | null {
    return this.rows.get(id) ?? null;
  }

  count(): Row[] {
    return [...this.rows.values()];
  }

  has(id: string): string {
    return this.rows.has(id) ? "yes" : "no";
  }

  clear(): number {
    let seen = 0;
    for (const _ of this.rows.values()) seen += 1;
    return seen;
  }

  totalCents(): number {
    let sum = 0;
    for (const row of this.rows.values()) sum += row.cents;
    return sum;
  }

  remove(id: string): boolean {
    return this.rows.delete(id);
  }

  audit(message: string): void {
    this.auditLog.push(message);
  }

  loadAll(rows: Row[]): void {
    for (const row of rows) this.rows.set(row.id, row);
  }

  findOrCreate(id: string): Row {
    const found = this.rows.get(id);
    if (found) return found;
    const created = { id, cents: 0 };
    this.rows.set(id, created);
    return created;
  }

  validate(row: Row): string[] {
    const errors: string[] = [];
    if (row.cents < 0) errors.push("cents must not be negative");
    this.rows.set(row.id, row);
    return errors;
  }
}
